app/src/main/java/emu/skyline/settings/SettingsActivity.kt

```diff
@@ -102,6 +109,69 @@ class SettingsActivity : AppCompatActivity() {
+                if (preference.key != null) {
+                    mapKeyTitle[preference.title.toString()] = preference
+                    noKey = false
+                }
+            }
+            if (!noKey)
+                mapKeyTitle[preferenceCategory.title.toString()] = preferenceCategory
+        }
+
+        menuInflater.inflate(R.menu.menu, menu)
+        val menuItem = menu!!.findItem(R.id.app_bar_search)
+        val searchView = menuItem.actionView as SearchView
+        searchView.queryHint = getString(R.string.search)
+
+        searchView.setOnQueryTextFocusChangeListener { _, focus ->
+            (binding.titlebar.toolbar.layoutParams as AppBarLayout.LayoutParams).scrollFlags =
+                if (focus)
+                    AppBarLayout.LayoutParams.SCROLL_FLAG_NO_SCROLL
+                else
+                    AppBarLayout.LayoutParams.SCROLL_FLAG_SCROLL
+        }
+
+        searchView.setOnQueryTextListener(object : SearchView.OnQueryTextListener {
+            override fun onQueryTextSubmit(query : String?) : Boolean {
+                return false
+            }
+
+            override fun onQueryTextChange(newText : String?) : Boolean {
+                val querys = newText!!.split(",").toTypedArray()
+                val filteredMap = mutableMapOf<String, Preference>()
+                for (query in querys) {
+                    filteredMap += mapKeyTitle.filterKeys { it.contains(query as CharSequence, true) }
+                }
+                if (newText.isNotEmpty() && filteredMap.isNotEmpty()) {
+                    (supportFragmentManager.findFragmentById(R.id.settings) as PreferenceFragmentCompat).preferenceScreen.forEach { preferenceCategory ->
+                        var allHiden = true
+                        (preferenceCategory as PreferenceCategory).forEach { preference ->
+                            preference.isVisible = filteredMap.containsValue(preference) || filteredMap.containsValue(preferenceCategory)
+                            if (preference.isVisible && allHiden) {
+                                allHiden = false
+                            }
+                        }
+                        // Hide PreferenceCategory if none of its preferences match the search and neither the category title
+                        preferenceCategory.isVisible = !allHiden || filteredMap.containsValue(preferenceCategory)
+                    }
+                } else { // If user input is empty or there's no match, show all preferences (including those without key)
+                    (supportFragmentManager.findFragmentById(R.id.settings) as PreferenceFragmentCompat).preferenceScreen.forEach { preferenceCategory ->
+                        preferenceCategory.isVisible = true
+                        (preferenceCategory as PreferenceCategory).forEach {
+                            it.isVisible = true
+                        }
+                    }
+                }
```

Comment:
In the case the input text is empty, all settings should be shown.
In the case the input text doesn't match anything, no preferences should be shown.